als-server/shared/src/main/scala/org/mulesoft/als/server/modules/workspace/StagingArea.scala

```diff
@@ -27,6 +21,13 @@ trait StagingArea[Parameter] extends SyncFunction {
   def dequeue(files: Set[String]): Unit =
     sync(() => files.foreach(pending.remove))
 
+  def dequeue(file: String, p: Parameter): Unit =
+    sync(() =>
+      pending.get(file) match {
+        case Some(param) if param equals p => pending.remove(file)
+        case _                             =>
```

Comment:
should we log a warning? or is this an expected outcome?

Reply:
Intended, only want to remove the file in the case the notification is the expected one